host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1448,7 +1594,30 @@ static MMAL_STATUS_T create_camera_component(RASPIVID_STATE *state)
    // Set the encode format on the video  port
 
    format = video_port->format;
-   format->encoding_variant = MMAL_ENCODING_I420;
+
+   if (state->raw_output)
+   {
+      switch (state->raw_output_fmt)
```

Comment:
I've changed the commit again. The video splitter is now connected between camera's output[0] and preview's input.